lib/Basics/memory-map-posix.cpp

```diff
@@ -76,28 +76,45 @@ int TRI_FlushMMFile(int fileDescriptor, void* startingAddress,
 
   int res = msync(startingAddress, numOfBytesToFlush, flags);
 
-#ifdef __APPLE__
-  if (res == TRI_ERROR_NO_ERROR) {
-    res = fcntl(fileDescriptor, F_FULLFSYNC, 0);
-  }
-#endif
+  if (res != TRI_ERROR_NO_ERROR) {
```

Comment:
This should probably be `if (res != 0) { // no error`, since it's a system call and no error is explicitly 0. While it's unlikely we'd ever change the definition of `TRI_ERROR_NO_ERROR` to be non-zero, it feels a little sketchy.